File: Serverside/Admin/Api/Dtos/Variant/CreateVariantDto.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace FNO.Admin.AdminApi.Dtos.Variant
{
    public class CreateVariantDto
    {
        [Required]
        public int ProductId { get; set; }
        [Required]
        public string ImagePath { get; set; }
        public decimal? DiscountPrice { get; set; }
        [Required]
        public List<int> ColorIds { get; set; }
        public List<CreateVariantInventoryInfoDto> InventoryInfos { get; set; }
    }
}

File: Clientside/Admin/src/models/CreateVariantDto.ts
import CreateVariantInventoryInfoDto from "./CreateVariantInventoryInfoDto";

interface CreateVariantDto {
	productId: number
	imagePath: string,
	discountPrice?: number,
	colorIds: number[],
	inventoryInfos: CreateVariantInventoryInfoDto[],
}

export default CreateVariantDto